extra_space.py

```diff
@@ -0,0 +1,51 @@
+# !/usr/bin/env python
+'''
+This module automatically checks for trailing spaces.
+It scans the complete folder, searches for python files
+(ignoring the virtual environment python files) and automatially
+removes the trailing spaces.
+'''
+
+import os
+import sys
+
+
+BASE_DIR = os.path.abspath(__file__)
+
+
+def fix_file(path):
+    '''
+    The function, using 'for' loop scans all the directories as well
+    as sub-directories of the folder for python file and checks
+    for trailing white space.
+    '''
+    p = ""
+    for dirpath, _, filenames in os.walk(path):
+        if(os.path.exists(os.path.join(dirpath, 'bin/activate'))):
+            p = dirpath
+        for f in filenames:
+            if f.endswith('.py'):
+                filename = os.path.abspath(os.path.join(dirpath, f))
+                if (not bool(p)):
```

Comment:
It took me a while to figure out what you meant there - setting `p = None` above and checking `if p is not None:` here would be more idiomatic.